fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/common/RecordSerializer.java

```diff
@@ -115,14 +115,28 @@ M deserialize(@Nonnull RecordMetaData metaData, @Nonnull Tuple primaryKey,
         DECRYPT_SERIALIZED_RECORD("decrypt serialized record");
 
         private final String title;
-        Events(String title) {
+        private final String logKey;
+
+        Events(String title, String logKey) {
             this.title = title;
+            this.logKey = logKey;
+        }
+
+        Events(String title) {
+            this(title, null);
         }
 
+
         @Override
         public String title() {
             return title;
         }
+
+        @Override
+        @Nonnull
+        public String logKey() {
+            return (this.logKey != null) ? this.logKey : StoreTimer.DetailEvent.super.logKey();
```

Comment:
I like that this abstracts the logic for getting the default log key, but it might make sense to memoize the result in the constructors for the ones we do have to avoid lower-casing the name each time (though as is puts us in s.q.a.b. w.r.t. performance, I suppose).

Reply:
Done. Please Review.